src/components/NcRichText/autolink.js

```diff
@@ -115,11 +115,26 @@ export const getRoute = (router, url) => {
 
 	const isAbsoluteURL = /^https?:\/\//.test(url)
 
+	// URL is a "mailto:", "tel:" or any custom app protocol link
+	const isNonHttpLink = /^[a-z][a-z0-9+.-]*:.+/.test(url)
+	if (!isAbsoluteURL && isNonHttpLink) {
+		return null
+	}
+
 	// URL is not a link to this Nextcloud server instance => not an app route
 	if (isAbsoluteURL && !url.startsWith(getBaseUrl())) {
 		return null
 	}
 
+	if (!isAbsoluteURL && !url.startsWith('/')) {
+		// Relative URL without trailing slash are not allowed
```

Comment:
You mean *leading* slash? What about real relative URLs like `./foo/bar`?

Reply:
Adjusted